### ISSUE TITLE
Suggestion: Appropriate optimizer for each BoTorch model class

### ISSUE THREAD

**Issue by YoungJaeBae** [facebook/Ax#1370]
I have noticed that ax is using `botorch.optim.optimize.optimize_acqf` for the optimization of models regardless of the surrogate model type. But, BoTorch recommends using a specific optimization function i.e. [MixedSingleTaskGP](https://botorch.org/api/models.html#module-botorch.models.gp_regression_mixed) is recommended to use `optimize_acqf_mixed`.

> A single-task exact GP model for mixed search spaces.

>This model is similar to SingleTaskGP, but supports mixed search spaces, which combine discrete and continuous features, as well as solely discrete spaces. It uses a kernel that combines a CategoricalKernel (based on Hamming distances) and a regular kernel into a kernel of the form

>>K((x1, c1), (x2, c2)) =K_cont_1(x1, x2) + K_cat_1(c1, c2) + K_cont_2(x1, x2) * K_cat_2(c1, c2)

>where xi and ci are the continuous and categorical features of the input, respectively. The suffix _i indicates that we fit different lengthscales for the kernels in the sum and product terms.

>Since this model does not provide gradients for the categorical features, optimization of the acquisition function will need to be performed in a mixed fashion, i.e., treating the categorical features properly as discrete optimization variables. We recommend using optimize_acqf_mixed.

It would be wonderful if AX supports various optimizer functions and enables the user to choose an optimizer or set a default optimizer adequate to the given BoTorch model.

**Comment by saitcakmak** [comment]
Hi @YoungJaeBae. I believe the acquisition optimization behavior changes based on what model / API you're using. We are transitioning towards the modular BoTorch model, and recommend that for more advanced use cases. In modular BoTorch setup, if you have discrete features (e.g. unordered categorical), it will use `optimize_acqf_mixed` or `optimize_acqf_discrete`. This is implemented [here](https://github.com/facebook/Ax/blob/main/ax/models/torch/botorch_modular/acquisition.py#L360).

For the legacy models (e.g. [here](https://github.com/facebook/Ax/blob/main/ax/models/torch/botorch_moo.py#L202)), we expose a `acqf_optimizer` kwarg , but I don't think we have a built in helper that uses the mixed optimizer. Implementing a custom wrapper around `optimize_acqf_mixed` and passing that in would be another option.

**Comment by YoungJaeBae** [comment]
Hi @saitcakmak. Thanks, a lot!  I shall give a test with a modular BoTorch setup
